File: fizzbuzz/fizz_buzz.c
#include <stdio.h>

int main(int argc, char *args[])
{
    int num;
    for (num = 1; num <= 100; num = num + 1) {
        if (num % 15 == 0) {
            printf("FizzBuzz\n");
        } else if (num % 3 == 0) {
            printf("Fizz\n");
        } else if (num % 5 == 0) {
            printf("Buzz\n");
        } else {
            printf("%d\n", num);
        }
    }
    return 0;
}

File: fizzbuzz/fizz_buzz.hs
fizz_buzz :: Integer -> String
fizz_buzz n
    | mod n 15 == 0 = "FizzBuzz"
    | mod n 3 == 0 = "Fizz"
    | mod n 5 == 0 = "Buzz"
    | otherwise = show n

main = do
    print $ [ fizz_buzz x | x <- [1..100] ]